### PR TITLE
Small changes NtupleTools README and submit_job.py.

### PR DESCRIPTION
**Pull request by kdlong**
Made a few small changes to submit_job.py (moved two distinct ways of reading datasets into separate functions, with comments), removed "export TERMCAP=screen\n", which isn't necessary unless someone is using screen (is anyone?), and added an option to specify an output file for the script to be written to.

Also made small changes to the README in NtupleTools/test

### CODE REVIEW

**Comment by dntaylor** [comment]
I use screen!

**Comment by kdlong** [comment]
But, is there any reason to have that in by default?

**Comment by kdlong** [comment]
My understanding is that it makes the output look better in screen, is that right? My thinking was that stuff like this can be confusing and doesn't serve a purpose for most people.

**Comment by dntaylor** [comment]
No idea. I don't really pay attention. I just use it when I'm at my home to avoid failures during submission if my internet craps out.

**Comment by dntaylor** [comment]
Actually, the datadefs stuff is obsolete since miniaod. Really, it can be deleted.

**Comment by kdlong** [comment]
Even better.

**Comment by dntaylor** [comment]
Add a commit where you just remove all the datadef parts and I'll check it tomorrow.

**Comment by gomber** [comment]
As far as I understand screen is used when you are running something interactively and doesn't want
it to be lost due to internet connection, so I do following :

> > screen
> > run your script
> > ctrl+a d (which detach your screen)

at this point you can close the terminal, close your laptop…

when you want it back again

reopen the terminal and do

> > screen -r

**Comment by kdlong** [comment]
Sounds good.

**Comment by dntaylor** [comment]
Kenneth, while you are doing all these updates/documentation, can you start adding it to a MiniAOD page on the wiki? https://github.com/uwcms/FinalStateAnalysis/wiki

**Comment by kdlong** [comment]
I think using screen is a very good idea, I use tmux for the same purpose. But, as I understand it, this screen-specific change to an environment variable only affects the appearance of the script when run from screen (someone can check me on this). I don't think it's necessary and I prefer to have the script stay as simple as possible with less stuff in it the average user doesn't need and doesn't understand.

**Comment by kdlong** [comment]
Thanks Devin, I'll try to make some edits there as well.

File: Utilities/scripts/submit_job.py [hunk 1]
```diff
@@ -25,8 +34,287 @@
...
+                command.append('firstRun=%i' % firstRun)
+            lastRun = sample_info.get('lastRun', -1)
+            if lastRun > 0:
+                command.append('lastRun=%i' % lastRun)
+
+        script_content += '# Submit file for sample %s\n' % dataset_name
+        script_content += 'mkdir -p %s\n' % os.path.dirname(dag_dir)
+        script_content += ' '.join(command) + '\n'
+    return script_content
+def datasets_from_datadefs(args):
```
Comment (dntaylor) [inline Utilities/scripts/submit_job.py]: @kdlong 
Do you want to delete this out as part of the removal of datadefs?

**Pull request merged**